src/api.rs

```diff
@@ -292,164 +333,229 @@ impl Context {
     sequence_header_inner(&self.seq).unwrap()
   }
 
-  pub fn frame_properties(&mut self, idx: u64) -> bool {
-    let key_frame_interval: u64 = self.fi.config.key_frame_interval;
+  fn setup_key_frame_properties(&mut self) {
```

Comment:
This might even be nicer as a fi initializer (e.g. new_key_frame or new_inter_frame). But you don't have to change it for this patch.